### PR TITLE
SCMOD-12238 : Handle unsupported mapping param additions and deletions

### PR DESCRIPTION
**Pull request by kusumaghoshdastidar**
Jira: https://portal.digitalsafe.net/browse/SCMOD-12238

Extended to handle unsupported param additions to field mapping which would have caused Elasticsearch mapping exceptions.

### CODE REVIEW

File: elastic-mapping-updater/src/main/java/com/github/cafdataprocessing/elastic/tools/ElasticMappingUpdater.java [hunk 1]
```diff
@@ -308,30 +360,6 @@ private static String getParamName(final String key)
...
-                                ((Map<?, ?>) (value.leftValue())).keySet()
-                        );
-                        //If one of the removed properties is unmodifiable, disable this update.
-                        if (!MODIFIABLE_PROPERTIES.containsAll(elementsRemoved)) {
-                            allowedFieldDifferences.remove(key);
-                        }
-                    });
-                unsupportedObjectChanges = true;
-            }
-
```
Comment (dermot-hardy) [inline elastic-mapping-updater/src/main/java/com/github/cafdataprocessing/elastic/tools/ElasticMappingUpdater.java:334]: Why is this section being removed?

File: elastic-mapping-updater/src/main/java/com/github/cafdataprocessing/elastic/tools/ElasticMappingUpdater.java [hunk 2]
```diff
@@ -386,7 +422,7 @@ private void removeUnsupportedFieldChange(final Map<String, Object> mappingsChan
         final List<String> path = Arrays.asList(StringUtils.split(fieldPath.trim(), "/"));
         final int size = path.size();
         int index = 0;
-        if(size == 2)
+        if(size <= 2)
```
Comment (dermot-hardy) [inline elastic-mapping-updater/src/main/java/com/github/cafdataprocessing/elastic/tools/ElasticMappingUpdater.java:425]: What is this change?

**Review by dermot-hardy — APPROVED** [review]

File: elastic-mapping-updater/src/main/java/com/github/cafdataprocessing/elastic/tools/ElasticMappingUpdater.java (hunk 1 above)
Comment (kusumaghoshdastidar) [reply elastic-mapping-updater/src/main/java/com/github/cafdataprocessing/elastic/tools/ElasticMappingUpdater.java:334]: Better handled when other unsupported changes are being detected in the isMappingChangeSafe function:
type change
adding un-modifiable params
removing un-modifiable params

File: elastic-mapping-updater/src/main/java/com/github/cafdataprocessing/elastic/tools/ElasticMappingUpdater.java (hunk 2 above)
Comment (kusumaghoshdastidar) [reply elastic-mapping-updater/src/main/java/com/github/cafdataprocessing/elastic/tools/ElasticMappingUpdater.java]: Not required...removed it.

**Review by dermot-hardy — APPROVED** [review]

**Pull request merged**